CONTRIBUTING.md

```diff
@@ -1,25 +1,70 @@
+### Issues
 
-Normally, the fields of the error struct are not public, as they can be accessed through the
-provided methods. However, in some cases, you may need to make the struct fields `pub` or
-`pub(crate)`.
+There are two major kinds of issues in Twilight, user-created and
+developer-created. Of the user-created issues, they are usually bug reports,
+feature requests, or support requests. If reporting a bug, please include the
+operating system used, any relevant information about the tech stack, and the
+feature flags used. Before making a support issue, be sure to consider joining
+the [Twilight Discord] and making a thread for your question in the `#support`
+channel.
 
-Any new error implementations must follow this pattern:
+Developer-created issues are most often bugs or tracking issues for supporting
+Discord API features. We aim for 100% coverage of the Discord API; we will wait
+until a new feature is supported in the [Discord API] before adding its support
+to Twilight.
+
+### Pull Requests
+
+In order to ease developer cooperation, please make a pull request from a
+feature branch on your own fork instead of from `main` or `next`. Additionally,
+since we squash every commit before merging, force-pushing a branch with new
+changes or updates is not always required. This helps with maintaining a
+chronological history in the PR story.
+
+Contributors must add tests and documentation that reflects their changes.
+
+Twilight follows a [Conventional Commit] style for pull request titles and
+commits. A Github Actions workflow automatically labels pull requests based on
+parsing the title. For more information, see [this discussion][#1439].
+
+Labels that must be added manually:
+- `d-api`: for changes that correspond with the [Discord API]
+- `d-breaking`: for API changes that cause a breaking change in library code
+- `d-unmerged`: changes that have not yet been merged into the documentation
+- `w-do-not-merge`: merge is blocked on another PR
+- `w-needs-more-docs`: a change needs more documentation in the code itself or
+  in the PR description
+- `w-needs-testing`: a change needs more tests added or proof of manual testing
+- `w-unapproved`: a design change has not yet been approved by library
+  maintainers
+
+Pull requests require two approvals before merging. The squashed commit
+description must contain a summary of the changes. 
+
+## Code Style
+
+### Errors
+
+Twilight's `Error` design is a struct with one required field (`kind`) and one
+optional field (`source`). It includes three methods (`kind`, `into_source`, and
+`into_parts`) which allow the user to access data within the error. The return
+types for any source errors are `dyn Error`, which allows us to update
+dependencies without breaking the public API.
+
+Normally, the fields of the error struct are not public, as they can be accessed
+through the provided methods. However, in some cases, you may need to make the
+struct fields `pub` or `pub(crate)`.
```

Comment:
They may be public in internal module errors, but public API error fields should *never* be public.